File: Day065_2025/Q065_Leetcode.java
class Q065_Leetcode {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        long sum = 0, sqrSum = 0;
        int n = grid.length;
        int total = n * n;

        for (int[] row : grid) {
            for (int num : row) {
                sum += num;
                sqrSum += (long) num * num;
            }
        }

        long sumDiff = sum - (long) total * (total + 1) / 2;
        long sqrDiff = sqrSum - (long) total * (total + 1) * (2 * total + 1) / 6;

        int repeat = (int) ((sqrDiff / sumDiff + sumDiff) / 2);
        int missing = (int) ((sqrDiff / sumDiff - sumDiff) / 2);

        return new int[]{repeat, missing};
    }
}

File: Day065_2025/Q065_Leetcode.py
from typing import List

class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        sum_val, sqr_sum = 0, 0
        n = len(grid)
        total = n * n

        for row in grid:
            for num in row:
                sum_val += num
                sqr_sum += num * num

        sum_diff = sum_val - (total * (total + 1) // 2)
        sqr_diff = sqr_sum - (total * (total + 1) * (2 * total + 1) // 6)

        repeat = (sqr_diff // sum_diff + sum_diff) // 2
        missing = (sqr_diff // sum_diff - sum_diff) // 2

        return [repeat, missing]
